numpy_ml/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np 
+
+class GaussianNB():
+    """
+    Gaussian Naive Bayes 
+
+    Assume each class conditional feature distribution is
+    independent and estimate the mean and variance from the
+    training data
+
+    Parameters
+    ----------
+    epsilon: float
+        a value that add to variance to prevent numerical error
+    
+    Attributes
+    ----------
+    num_class : ndarray of shape (n_classes,)
+        count of each class in the training sample
+
+    mean: ndarray of shape (n_classes,)
```

Comment:
I don't think these dimensions are quite right. Shouldn't `mean` and `sigma` each be arrays of dimension`(n_classes, n_features)`? That is, we expect a single univariate feature mean/variance for _each_ feature dimension.